workspace/exercise_test.go

```diff
@@ -17,9 +17,9 @@ func TestHasMetadata(t *testing.T) {
 	exerciseA := Exercise{Root: ws, Track: "bogus-track", Slug: "apple"}
 	exerciseB := Exercise{Root: ws, Track: "bogus-track", Slug: "banana"}
 
-	err = os.MkdirAll(filepath.Join(exerciseA.Filepath()), os.FileMode(0755))
+	err = os.MkdirAll(filepath.Join(exerciseA.Filepath(), ignoreSubdir), os.FileMode(0755))
```

Comment:
Tests having to know to create the `.exercism` ignore subdirectory seems smelly. I'm not sure how to avoid it though.

Reply:
Good call. I think I fixed this up in all the tests that were explicitly knowing about metadata filepaths.